spec/workflow/runner/docker_spec.rb

```diff
@@ -0,0 +1,42 @@
+RSpec.describe ManageIQ::Floe::Workflow::Runner::Docker do
+  require "awesome_spawn"
+
+  let(:subject) { described_class.new }
+  let(:result)  { AwesomeSpawn::CommandResult.new(nil, nil, nil, 0) }
+
+  describe "#run!" do
+    it "raises an exception without a resource" do
+      expect { subject.run!(nil) }.to raise_error(ArgumentError, "Invalid resource")
+    end
+
+    it "raises an exception for an invalid resource uri" do
+      expect { subject.run!("arn:abcd:efgh") }.to raise_error(ArgumentError, "Invalid resource")
+    end
+
+    it "calls docker run with the image name" do
+      expect(AwesomeSpawn)
+        .to receive(:run!)
+        .with("docker", :params => array_including("run", :rm, "hello-world:latest"))
+        .and_return(result)
```

Comment:
There are built-in helpers in AwesomeSpawn for this (see https://github.com/ManageIQ/awesome_spawn/blob/master/lib/awesome_spawn/spec_helper.rb)

Reply:
Implemented this in b9d598ad3c24876d225dd2e8f5a0a7c0e155d0ab